src/lib/Frontend/Var.cpp

```diff
@@ -66,63 +69,40 @@ static Var &cmpOp(const Var &L, const Var &R, IntOp IOp, FPOp FOp) {
   return ResultVar;
 }
 
-Var::Var(AllocaInst *Alloca, FuncBase &Fn, Type *PointerElemType)
-    : Alloca(Alloca), Fn(Fn), PointerElemType(PointerElemType) {}
+Var::Var(AllocaInst *Alloca, FuncBase &Fn) : Alloca(Alloca), Fn(Fn) {}
 
-Value *Var::getValue() const {
-  auto &IRB = Fn.getIRBuilder();
-  Type *AllocaType = Alloca->getAllocatedType();
-  if (AllocaType->isPointerTy()) {
-    auto *Ptr = IRB.CreateLoad(AllocaType, Alloca);
-    return IRB.CreateLoad(PointerElemType, Ptr);
-  }
-  return IRB.CreateLoad(AllocaType, Alloca);
+Var::Var(Value * /*PointerValue*/, FuncBase &Fn) : Alloca(nullptr), Fn(Fn) {}
```

Comment:
Where is this constructor used?

Reply:
> Where is this constructor used?

It's used by the [ArrayVar constructor](https://github.com/Olympus-HPC/proteus/blob/feature/frontend-array/src/lib/Frontend/Var.cpp#L839)